### PR TITLE
Add flaky test detection github workflow

### PR DESCRIPTION
**Pull request by dimkouv**
Run flaky test detection on a schedule and reduce test `-count` to `1` for normal test runs.

### CODE REVIEW

File: .github/workflows/ccip-find-flaky-tests.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        run: TEST_COUNT=50 make test
+      - name: Notify the team by commenting on latest commit
+        if: failure()
+        run: |
+          COMMIT_SHA=$(git rev-parse HEAD)
+          curl -X POST \
+            -H "Authorization: token ${{ secrets.GITHUB_TOKEN }}" \
+            -H "Accept: application/vnd.github.v3+json" \
+            https://api.github.com/repos/${{ github.repository }}/commits/$COMMIT_SHA/comments \
+            -d '{"body":"@smartcontractkit/ccip-offchain flaky tests were detected, please check."}'
```
Comment (winder) [inline .github/workflows/ccip-find-flaky-tests.yml:37]: What commit is this? I think we need slack integration.

Comment (mateusz-sekara) [reply .github/workflows/ccip-find-flaky-tests.yml:37]: 👍 for slack notification, there is too much spam from Github already

Comment (makramkd) [reply .github/workflows/ccip-find-flaky-tests.yml]: I don't have the ability anymore, we need to create a security ticket

File: .github/workflows/ccip-find-flaky-tests.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    steps:
+      - uses: actions/checkout@9bb56186c3b09b4f86b1c65136769dd318469633 # v4.1.2
+      - name: Setup Go ${{ matrix.go-version }}
+        uses: actions/setup-go@0c52d547c9bc32b1aa3301fd7a9cb496313a4491 # v5.0.0
+        with:
+          go-version: ${{ matrix.go-version }}
+      - name: Display Go version
+        run: go version
+      - name: Run tests 50 times.
+        run: TEST_COUNT=50 make test
```
Comment (makramkd) [inline .github/workflows/ccip-find-flaky-tests.yml:28]: Is 50 enough to detect flakes? I guess if this runs many times a day we might get lucky.

**Review by CL-Andrew — APPROVED** [review]

**Pull request closed without merging**